### ISSUE TITLE
Web - Home Page - Missing Heading

### ISSUE THREAD

**Issue by jfc3-dol** [USDepartmentofLabor/Comply-Chain#66]
There are no programmatic headings on the page only what look like visual ones.

Home Page - https://dolcontentdev.opadev.dol.gov/ilab/complychain/build/

**Comment by KyleLivingstonDOL** [comment]
Hi John, can you explain the difference between programmatic versus visual headings? Where are these headings/where should these headings be? Thanks!
Kyle

**Comment by jfc3-dol** [comment]
Visual headings are what a person sees, and programmatic headings are ones that are coded using a heading tag, for example of `<h2>Heading two</h2>`.

A visual heading is one that the text is large an bold but not coded as a heading. Screen reader users using programmatic headings can skip from one heading to the next by using the "H" key or the number of the heading.

**Comment by trevitalgit** [comment]
I see <h1> tags in the headings. I tested with voice over and it reads the headings. Please advise what else I need to do with this ticket.
![image](https://user-images.githubusercontent.com/53573580/63095781-14d4a780-bf3a-11e9-9300-7a8aed3ff127.png)

**Comment by jfc3-dol** [comment]
Each of the buttons should not be a heading level one (h1). The web best practice is to have one heading level one (h1) per page. So all those should be a heading level two (h2) because they would fall under whatever the heading level one (h1)  would be. There should be some text that tells the user what application they are on. The headings apply to the iOS and Android applications too.

**Comment by KyleLivingstonDOL** [comment]
> Each of the buttons should not be a heading level one (h1). The web best practice is to have one heading level one (h1) per page. So all those should be a heading level two (h2) because they would fall under whatever the heading level one (h1) would be. There should be some text that tells the user what application they are on. The headings apply to the iOS and Android applications too.

Thanks John. Appreciate the best practice. Is this a 508 issue or a best practice? I believe the reason we have these as H1 is because they have sub-headings underneath them in subsequent pages. Is the screen reader mis-reading these in some way that would affect 508 compliance?

**Comment by jfc3-dol** [comment]
@KyleLivingstonDOL,

The name of the application "Comply Chain" should be the Heading level one on the main screen with those others to be a heading level two (h2) and any others under that being a heading level three (h3).

Think of it as a book were there is only one TITLE, so it would be a heading level one (h1). Any of the chapters under the title would be a heading level two (h2). With subtopics being heading level three (h3), etc.

**Comment by transreductionist** [comment]
Hi @jfc3-dol and @KyleLivingstonDOL,

After fixing the defect I came here to get some more information and saw that I came close to the fix. I named the page "Comply Chain Homepage". Would you like that to be just "Comply Chain"? This is a title at <h1>. 

As requested all items have been assigned the tag <h2>.

Take a look at the screenshot and let me know if this layout is okay.

![Screenshot 2020-02-13 11 22 48](https://user-images.githubusercontent.com/8593071/74455437-4d0a6380-4e53-11ea-8931-cf196a4207de.png)

**Comment by jfc3-dol** [comment]
@transreductionist,
Yes, it's better to be Comply Chain" on the main page instead of "Comply Chain homepage".

**Comment by jfc3-dol** [comment]
Each of the topics on the home page is a heading level two (h2)  as expected.

I do not see a heading level one (h1) item for the page, and that is even if it was off-screen for screen reader and other assistive technology users.

**Comment by Abychalil** [comment]
What <h1> heading should be given on the home page? If we give "Comply Chain", there is already another text "Comply Chain" on the left top side. If that's fine, we can give comply chain itself as h1, instead of 'Comply Chain Homepage' as shown in the above screenshot. 
@jfc3-dol

**Comment by Abychalil** [comment]
What heading should be given on the home page? If we give "Comply Chain", there is already another text "Comply Chain" on the left top side. If that's fine, we can give comply chain itself as h1, instead of 'Comply Chain Homepage' as shown in the above screenshot.

@jfc3-dol

**Comment by jfc3-dol** [comment]
@Abychalil 

On the Home page, I do not see the text "Comply Chain Homepage", at least when I'm there the first time. You are right; it should be a heading level one (1h). I would think for ONLY the Home page. You could make the text "Comply Chain" that is a link back to itself to be the heading level one (h1) instead of adding "Comply Chain Homepage".

Does that make sense the way I explained it?

I have noticed it show up before depending on which other page I come from the get back there.

**Comment by jfc3-dol** [comment]
@harpreetsdev @Abychalil @pbhatt17

I do not see heading level one (h1) on the home page with any of my testing tools, so going to leave this open.

Maybe the update is coming in the next sprint.

**Comment by harpreetsdev** [comment]
@jfc3-dol 

Where do you want the new Comply chain heading to be situated, since there is already one on the left that says Comply chain? Or instead of adding another heading that says the same thing we could convert the existing heading into a H1 heading for accessibility purposes?

Let me know what you think.

@pbhatt17 @Abychalil

**Comment by jfc3-dol** [comment]
@harpreetsdev @pbhatt17 @Abychalil

Not quite sure where to put the new heading level one (h1) since you already have the text "Comply Chain" as a link back to the home page. If you added the text there again, it might be confusing to lots of people.

I'm thinking, is it possible to make the link text "Comply Chain" a heading level one (h1) only on the home page banner/header section? If so, that might be the answer to our issue. But if it's used on other pages /screen, that would end up giving the page two heading level one (h1) items which we don't want.

What do you all think?

**Comment by harpreetsdev** [comment]
@jfc3-dol 

I feel that would be a good idea to assign the existing heading "h1" tag. I feel it shouldn't be an issue. Let me try this as a fix, if there are multiple h1 headings on the page then we'll revert back. If it works we'll send it for your review.

@pbhatt17 @Abychalil

**Comment by jfc3-dol** [comment]
@harpreetsdev @pbhatt17 @Abychalil

Thanks, I look forward to testing the update in the next release.

**Comment by jfc3-dol** [comment]
@harpreetsdev @pbhatt17 @Abychalil

The Home Page now has a heading level one (h1) of "Comply Chain", which is the main navigation link—followed by heading level two (h2) items for the different Steps. I checked the iOS version, and the same applies to it. I  assume the Android version is the same. I Will test once I get the latest application.

So I'm going to close this issue.

Please see the attached screenshot.

![ilab - comply chain - home page - The Home Page now has a heading level one (h1) as the main navigation link  Followed by heading level two (h2) items for the different Steps](https://user-images.githubusercontent.com/10423487/107054868-a70be380-679e-11eb-9b0b-07ab0cda2e05.png)
